src/main/java/com/dsrm/dsrmbackend/services/impl/RoomServiceImpl.java

```diff
@@ -18,17 +19,38 @@
 public class RoomServiceImpl implements RoomService {
     private final RoomRepo repository;
     private final RoomMapper roomMapper;
+    private final RoomTypeMapper roomTypeMapper;
     public Room addRoom(RoomRequestDTO roomDto) {
         Room room = roomMapper.roomReqDTOToRoom(roomDto);
         room.setId(UUID.randomUUID().toString());
         return repository.save(room);
     }
 
-    public Optional<Room> getRoom(String roleId) {
-        return repository.findById(roleId);
+    public Optional<Room> getRoom(String roomId) {
+        return repository.findById(roomId);
     }
 
     public Page<Room> getRooms(Pageable pageable) {
         return repository.findAll(pageable);
     }
+
+    @Override
+    public Room updateRoom(RoomRequestDTO update, String roomId) {
+        Optional<Room> room = repository.findById(roomId);
+        Room newRoom;
+        if (room.isPresent()) {
```

Comment:
I believe we should only update fields if they are not null. And also in both cases when you want to add roomType you have to fetch it from database instead of using mapper. It is very dangerous because you can create new one event if we don't want to. Look at com.dsrm.dsrmbackend.services.impl.UserServiceImpl#updateUserRoles . You should set room type in the same way as the roles are set here.

Reply:
Done but github hasn't marked it